docs: use deep outline for typebox sidebar

marshallswain opened the pull request:
Having the deep outline makes it much easier to navigate the massive TypeBox API. Mobile devices only see a short table of contents, but for larger screens, it’s much easier to see what’s available when the sidebar shows the full tree. Also, there’s so much in the document that it’s easy to get lost.  The deep outline solves that.

![Screenshot 2022-12-29 at 7 38 37 PM](https://user-images.githubusercontent.com/128857/210028976-0a2b7bb6-4839-4a3f-85ce-321a7ba1c65d.jpg)

marshallswain committed:
docs: use deep outline for typebox sidebar

Having the deep outline makes it much easier to navigate the massive TypeBox API. Mobile devices only see a short table of contents, but for larger screens, it’s much easier to see what’s available when the sidebar shows the full tree. Also, there’s so much in the document that it’s easy to get lost.  The deep outline solves that.

daffl commented:
I think I changed this because it was pushing down important information like the query schema and validator docs. Is there a way to define the level of nesting?

marshallswain commented:
@daffl There is a way, but it's broken and puts all levels into a single list without indentation.  I think we need to reorder the page to put the types last.

marshallswain committed:
docs: place types last in typebox docs

the types were pushing other important content down out of sight.  Putting types last keeps everything else visible.

marshallswain commented:
@daffl, I've updated the page to put the Types last, so they don't push down other important stuff.

marshallswain commented:
I've updated the screenshot.

daffl commented:
That works. Thanks!
